test/tests/functional/pbs_test_tpp.py

```diff
@@ -50,72 +50,184 @@ class TestTPP(TestFunctional):
     def setUp(self):
         TestFunctional.setUp(self)
         self.pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        self.pbs_conf_path = self.du.get_pbs_conf_file(
+            hostname=self.server.hostname)
+        msg = "Unable to retrieve pbs.conf file path"
+        self.assertNotEqual(self.pbs_conf_path, None, msg)
+
         self.exec_path = os.path.join(self.pbs_conf['PBS_EXEC'], "bin")
         if not self.default_client:
             self.default_client = self.server.client
 
-    def submit_job(self, set_attrib=None, exp_attrib=None, sleep=10,
-                   job_script=False, interactive=False):
+    def pbs_restart(self, host_name):
+        """
+        This function starts PBS daemons after updating pbs.conf file
+        :param host_name: Name of the host on which PBS
+                          has to be restarted
+        :type host_name: String
+        """
+        pi = PBSInitServices(hostname=host_name)
+        pi.restart()
+
+    def set_pbs_conf(self, host_name, conf_param):
+        """
+        This function sets attributes in pbs.conf file
+        :param host_name: Name of the host on which pbs.conf
+                          has to be updated
+        :type host_name: String
+        :param conf_param: Parameters to be updated in pbs.conf
+        :type conf_param: Dictionary
+        """
+        pbsconfpath = self.du.get_pbs_conf_file(hostname=host_name)
+        self.du.set_pbs_config(hostname=host_name, fin=pbsconfpath,
+                               confs=conf_param)
+        self.pbs_restart(host_name)
+
+    def unset_pbs_conf(self, host_name, conf_param):
+        """
+        This function unsets parameters in pbs.conf file
+        :param host_name: Name of the host on which pbs.conf
+                          has to be updated
+        :type host_name: String
+        :param conf_param: Parameters to be removed from pbs.conf
+        :type conf_param: List
+        """
+        pbsconfpath = self.du.get_pbs_conf_file(hostname=host_name)
+        self.du.unset_pbs_config(hostname=host_name,
+                                 fin=pbsconfpath, confs=conf_param)
+        self.pbs_restart(host_name)
+
+    def submit_job(self, job=False, jset_attr=None, jexp_attr=None,
+                   sleep=10, job_script=False, interactive=False,
+                   iset_attr=None, iexp_attr=None, resv_job=False,
```

Comment:
can we say job_set_attr and job_exp_attr.
Also, we can reduce the number of parameters here by using jset_attr and jexp_attr for interactive and reservation jobs as well. what do you think?